scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/JobDeadlineEmailNotification.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.policy.edf;
+
+import static org.ow2.proactive.scheduler.policy.edf.EDFPolicyExtended.getEffectiveDeadline;
+import static org.ow2.proactive.scheduler.policy.edf.EDFPolicyExtended.getEffectiveExpectedExecutionTime;
+
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.addons.email.exception.EmailException;
+import org.ow2.proactive.scheduler.common.JobDescriptor;
+import org.ow2.proactive.scheduler.core.JobEmailNotificationException;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.scheduler.descriptor.JobDescriptorImpl;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.util.SendMail;
+
+;
+
+
+public class JobDeadlineEmailNotification {
+
+    public static final String GENERIC_INFORMATION_KEY_EMAIL = "EMAIL";
+
+    private static final Logger LOGGER = Logger.getLogger(JobDeadlineEmailNotification.class);
+
+    private SendMail sender;
+
+    private List<JobDescriptor> jobs;
+
+    private static final Logger logger = Logger.getLogger(JobDeadlineEmailNotification.class);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename field "logger" to prevent any misunderstanding/clash with field "LOGGER" defined on line 51. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1845)